Store.js

```diff
@@ -140,14 +151,130 @@ define(["dcl/dcl", "decor/Invalidating"], function (dcl, Invalidating) {
 						collection.on("delete", this._itemRemoved.bind(this));
 						collection.on("refresh", this._refreshHandler.bind(this));
 					}
-					return this.processCollection(collection);
+				} else {
+					this.itemHandles = [];
+					this.observeCallbackArray = this._observeCallbackArray.bind(this);
+					this.observeCallbackItems = this._observeCallbackItems.bind(this);
+					for (var i = 0; i < this.store.length; i++) {
+						// affect the callback to the observe function if the item is observable
+						this.itemHandles[i] = Observable.observe(this.store[i], this.observeCallbackItems);
+					}
+					collection = processQueryResult.call(this, this.store.filter(this._isQueried, this));
+					// affect the callback to the observe function if the array is observable
+					this.storeHandle = ObservableArray.observe(this.store, this.observeCallbackArray);
```

Comment:
If you think it is better not to, I will modify the name with an underscore as for the observecallbacks. Maybe should I do it for the itemHandlers too ?